### PR TITLE
99 - Sidebar rules

### LINKED ISSUE #99: Sidebar Rules
The cards for the Sidebar sections have the following rules for all pages, except for home page:
*  **Trends and Research** section: Latest 3 blog articles that have the **Trend: Trends and Research** metadata in place.
*  **Featured** section: Latest 3 blog articles, except what is already in Trends and Research.

The home page sidebar must remain manually editable, so authors can individually decide what should appear there

The sidebar should be auto blocked on all other pages, except for the homepage.

The sidebar has the following order of items on Topic/Category/Year/Author pages:
* Featured article cards
* Trends and Research cards
* Topics list
* Years list

The sidebar has the following order of items on the Blog Article pages:
* Topics list
* Featured article cards
* Trends and Research cards
* Years list

### PR DESCRIPTION
**Pull request by andreituicu**
The following PR:
* Autoblocks for Category/Year/Topic/Author pages the locale specific sidebar from `/fragments/sidebar-common-fragment`
* Autoblocks for Articles the locale specific sidebar from `/fragments/sidebar-article-fragment`
* Implements the rules for Featured and Trends & Research cards
* Still eager loads the fragment it sidebar as it is critical for the layout and we want to make sure we don't cause CLS
* Defers the loading of subblocks from the sidebar fragment to prioritise the FCP and LCP blocks first

Fix #99 

Test URLs:
Homepage (manual edited inlined sidebar, not auto-blocked)
- Before: https://main--servicenow--hlxsites.hlx.live/blogs
- After: https://sidebar-rules--servicenow--hlxsites.hlx.live/blogs

Blog Article
- Before: https://main--servicenow--hlxsites.hlx.live/blogs/2023/demo/maximize-efficiency-hyperautomation
- After: https://sidebar-rules--servicenow--hlxsites.hlx.live/blogs/2023/demo/maximize-efficiency-hyperautomation

Blog Topic
- Before: https://main--servicenow--hlxsites.hlx.live/blogs/topics/ai-and-automation
- After: https://sidebar-rules--servicenow--hlxsites.hlx.live/blogs/topics/ai-and-automation

Blog Category
- Before: https://main--servicenow--hlxsites.hlx.live/blogs/category/company-news
- After: https://sidebar-rules--servicenow--hlxsites.hlx.live/blogs/category/company-news

Blog Year
- Before: https://main--servicenow--hlxsites.hlx.live/blogs/2023
- After: https://sidebar-rules--servicenow--hlxsites.hlx.live/blogs/2023

Blog Author
- Before: https://main--servicenow--hlxsites.hlx.live/blogs/author/mary-hain
- After: https://sidebar-rules--servicenow--hlxsites.hlx.live/blogs/author/mary-hain

Blog Article (FR)
- Before: https://sidebar-rules--servicenow--hlxsites.hlx.live/fr/blogs/2023/pourquoi-il-faut-preparer-les-employes-a-l-avenir-du-travail
- After: https://sidebar-rules--servicenow--hlxsites.hlx.live/fr/blogs/2023/pourquoi-il-faut-preparer-les-employes-a-l-avenir-du-travail

Blog Topic (FR)
- Before: https://main--servicenow--hlxsites.hlx.page/fr/blogs/topics/careers
- After: https://sidebar-rules--servicenow--hlxsites.hlx.live/fr/blogs/topics/careers

### CODE REVIEW

**Comment by sergiucoman** [comment]
As this is locale specific, could you add an example in a different language so that we check how it works.

File: blocks/cards/cards.js [hunk 1]
```diff
@@ -76,22 +78,63 @@ function fetchAPIBasedCards(cardInfos, apis) {
...
-      cardInfos[idx] = FILTERS.category(
-        // skip first 3 blogs as they are used for the latest section
-        blogs.slice(3),
-        toClassName(config.category),
-      ).slice(0, 3);
+function homepageLatestRule(blogs, cardInfos, idx) {
+  cardInfos[idx] = blogs.slice(0, 3);
+}
+
+function homepageCategoryRule(blogs, cardInfos, idx, config) {
```
Comment (sergiucoman) [inline blocks/cards/cards.js:85]: Is there a scenario where the top 3 articles are not the most "recent" one? I am thinking someone correcting an article that has initially been published in the past.

Comment (sergiucoman) [reply blocks/cards/cards.js:85]: The authoring experience looks good to me and we don't have to change it. But, as an alternative we could potentially consider the following experience: 

+---------------------------+
| Cards (Sidebar, Featured)           |
+--------+------------------+

### Trends & Research

+--------------------------------------+
| Cards (Sidebar, Trends and Research )                     
+--------+-----------------------------+

**Review by sergiucoman — APPROVED** [review]

File: blocks/cards/cards.js (hunk 1 above)
Comment (andreituicu) [reply blocks/cards/cards.js:85]: @sergiucoman 
> Is there a scenario where the top 3 articles are not the most "recent" one? I am thinking someone correcting an article that has initially been published in the past.

my understanding is that no, that scenario cannot happen.

>isn't the index sorted by publication date at any rate (and that's manually maintained and doesn't change when an article is corrected)?

It's how @benpeter mentions here, if I understood correctly. The initial publication date metadata remains unchanged on corrections. What they do is add a `Updated ${date of update}` in the text to mark corrections. I've found this example: https://www.servicenow.com/blogs/2020/covid-business-resources.html

Comment (andreituicu) [reply blocks/cards/cards.js:85]: > The authoring experience looks good to me and we don't have to change it. But, as an alternative we could potentially consider the following experience

That's true, we could've opted for that experience as well. I tried to keep the authoring consistent with the blog list and the cards on the homepage, which require more configuration than just the rule itself.

the way that I approach it in my mind is:
* block variation - what is in parentheses Cards (Sidebar) - I use when I need to write different CSS based on the class that gets generated .sidebar
* block configuration - I use when I need the content from the block to change, but styling to remain consistent

In the end, both experiences are equally valid from my point of view, just that having already implemented two instances of this one, I tried to make it consistent.

**Comment by andreituicu** [comment]
> As this is locale specific, could you add an example in a different language so that we check how it works.

Very good point! Added and example and tested. Thank you!

**Pull request merged**